src/main/java/com/solvd/model/Product.java

```diff
@@ -15,8 +16,55 @@ public class Product {
     private Integer rating;
     private Integer reviewNumber;
     private Boolean inStock;
-    // TODO add avaliable colors and sizes
+    // TODO add available colors and sizes
 
+    /**
+     * Returns true, if there are any different field values
+     * for filed pairs that are both not null. If other is null, it will return true.
+     * <p>
+     * I.e. if this.name is "foo" && other.name is "bar" it will return true,
+     * but if this.name is "foo" && other.name is null it will return false.
+     * (assuming there are no other differences),
+     * Warning: this method does not constitute an equivalence relationship
+     */
+    public boolean hasDifferences(Product other) {
+        if (other == null) {
+            return true;
+        }
+        if (differsInValue(getName(), other.getName())) return true;
+        if (differsInValue(getPrice(), other.getPrice(), (a, b) -> a.compareTo(b) == 0)) return true;
+        if (differsInValue(getRating(), other.getRating())) return true;
+        if (differsInValue(getReviewNumber(), other.getReviewNumber())) return true;
+        if (differsInValue(getInStock(), other.getInStock())) return true;
```

Comment:
could you please avoid such big count of if()

Reply:
Resolved in commit d7e8b1977c4216d47a46b008acd92320002294be